### PR TITLE
improve deprecation messages

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php [hunk 1]
```diff
@@ -101,7 +101,7 @@ public function getController(Request $request)
      */
     public function getArguments(Request $request, $controller)
     {
-        @trigger_error(sprintf('%s is deprecated as of 3.1 and will be removed in 4.0. Implement the %s and inject it in the HttpKernel instead.', __METHOD__, ArgumentResolverInterface::class), E_USER_DEPRECATED);
+        @trigger_error(sprintf('The "%s()" method is deprecated as of 3.1 and will be removed in 4.0. Implement the %s and inject it in the HttpKernel instead.', __METHOD__, ArgumentResolverInterface::class), E_USER_DEPRECATED);
```
Comment (Tobion) [inline src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php]: in other places is says "since Symfony x.x"

**Comment by Tobion** [comment]
Same as #27860 it does not seem worth to create so many conflicts.

**Comment by xabbuh** [comment]
I agree with the reasoning in #27860. Though I think things are different when it comes to deprecation messages as they are dropped on the next major version.

**Comment by nicolas-grekas** [comment]
The deprecation in the constructor of `TranslatorPass` is missing context, it doesn't tell about which class is concerned, leading to "where is this comming from" effect read standalone.

Since this is proposed on the lowest branch, the merge conflicts will happen only once, so that's OK.

**Comment by xabbuh** [comment]
I am not sure I understand what you mean. The `TranslatorPass` is not changed here.

**Review by nicolas-grekas — APPROVED** [review]
(please merge 3.4 into 4.0 just after merging this PR)

**Pull request merged**